sherpa/astro/ui/utils.py

```diff
@@ -1841,14 +1870,35 @@ def load_pha(self, id, arg=None, use_errors=False):
 
         phasets = self.unpack_pha(arg, use_errors)
 
-        if numpy.iterable(phasets):
-            num = len(phasets)
-            for id, pha in enumerate(phasets):
-                self.set_data(id + 1, pha)
-            if num > 1:
-                info("Multiple data sets have been input: 1-%s" % num)
-        else:
+        if not numpy.iterable(phasets):
             self.set_data(id, phasets)
+            return
+
+        # One issue with the following is that if there's
+        # only one file and id is a string then the
+        # output will be "foo1" rather than "foo" (when
+        # id="foo").  DJB thinks we can live with this.
```

Comment:
I think we can call that a feature and not a bug!

Reply:
I just wanted to note this, as it is subtly different to `load_pha('goo', singlefile)` when singlefile is not a PHA2 container, since this just creates a dataset called `goo`.

Reply:
It becomes relevant if the want to make a further enhancement (not for this PR), that allows selecting the PHA rows to read. For HETGS we typically extract 12 rows (HEG and MEG from -3 to +3), but if I already know I'm looking at a weak and relatively soft source, I only need MEG-1 and MEG+1. However, for now, I don't think that's a feature that's important to have because (a) the memory demand is low and (b) people can manipulate the PHAII file outsode of sherpa (e.g. tell tg_extract which orders they want). This might become more important in the future. Arcus will have 4 channels * 20 orders = 80 spectra or peopel could use PHAII file to store, e.g. all ACIS spectra from 154 identified sources in the FOC of an observation. However, I would argue that that's their problem. Sherpa just reads "one file" and if you want less, make a smaller file.

Reply:
I had briefly thought about being able to specify the ids in some way - e.g. you could name them "megp1", "megm1", ... - but that is a "big" change to how we think about ids, whereas this PR is pretty tiny ;-)

Reply:
That's what I do by hand, i.e. using `copy_data`, but that's for a future PR (which is not hard to build on top of this one.) So far, Sherpa is pretty generic, but I can imagine something like:
```
naming_functions = {'CHANDRA': chandra_namer,
                                  ...}
def chandra_namer(pha2):
    return ['{}_{}'.format(['zero','heg', 'meg','leg'][row['TGPART']],row['TGM']) for row in pha2]
```
and in the load function:
```
try:
    ids = nameing_functions[pha2.meta['TELESCOP']](pha2)
except:
   ids = ... what's in this PR because that works for all instruments ...
```
Not sure if that has to be backend dependent or if there is a way to iterate over rows in a backend-independent way.